File: 101-symmetric-tree/101-symmetric-tree.java

class Solution {
    public boolean isSymmetric(TreeNode r) {
         return isSymmetric(r, r);
  }
  private boolean isSymmetric(TreeNode m, TreeNode n) {
    if (m == null || n == null)
      return m == n;
    return m.val == n.val && isSymmetric(m.left, n.right) && isSymmetric(m.right, n.left);
  }
}
    

File: GO-Leet/101. Symmetric Tree.go
package main

func isSymmetric(root *TreeNode) bool {
	return root == nil || help(root.Left, root.Right)
}

func help(left *TreeNode, right *TreeNode) bool {

	if left == nil || right == nil {
		return left == right
	}

	if left.Val != right.Val {
		return false
	}
	return help(left.Left, right.Right) &&
		help(left.Right, right.Left)
}
